packages/material-ui/src/Box/styleFunction.js

```diff
@@ -0,0 +1,119 @@
+import {
+  borders,
+  display,
+  flexbox,
+  grid,
+  positions,
+  palette,
+  shadows,
+  sizing,
+  spacing,
+  typography,
+} from '@material-ui/system';
+
+const filterProps = [
+  ...borders.filterProps,
+  ...display.filterProps,
+  ...flexbox.filterProps,
+  ...grid.filterProps,
+  ...positions.filterProps,
+  ...palette.filterProps,
+  ...shadows.filterProps,
+  ...spacing.filterProps,
+  ...typography.filterProps,
+  'sx',
+];
+
+const getThemeValue = (prop, value, theme) => {
+  const inputProps = {
+    [prop]: value,
+    theme,
+  };
+
+  if (borders.filterProps.indexOf(prop) !== -1) {
+    return borders(inputProps);
+  }
+  if (display.filterProps.indexOf(prop) !== -1) {
+    return display(inputProps);
+  }
+  if (flexbox.filterProps.indexOf(prop) !== -1) {
+    return flexbox(inputProps);
+  }
+  if (grid.filterProps.indexOf(prop) !== -1) {
+    return grid(inputProps);
+  }
+  if (positions.filterProps.indexOf(prop) !== -1) {
+    return positions(inputProps);
+  }
+  if (palette.filterProps.indexOf(prop) !== -1) {
+    return palette(inputProps);
+  }
+  if (shadows.filterProps.indexOf(prop) !== -1) {
+    return shadows(inputProps);
+  }
+  if (sizing.filterProps.indexOf(prop) !== -1) {
+    return sizing(inputProps);
+  }
+  if (spacing.filterProps.indexOf(prop) !== -1) {
+    return spacing(inputProps);
+  }
+  if (typography.filterProps.indexOf(prop) !== -1) {
+    return typography(inputProps);
+  }
+  return { [prop]: value };
+};
+
+const traverseSx = (styles, theme) => {
+  if (!styles) return null;
+
+  if (typeof styles !== 'object') {
```

Comment:
Should we support a function to pull value from the theme on each property?

Reply:
This is done now